### ISSUE TITLE
Sentinel Unable To Connect to Daemon (testnet 12.1 Ubuntu)

### ISSUE THREAD

**Issue by ryansd92** [dashpay/dash#1108]
One of my testnet nodes that was running yesterday's build transitioned to WATCHDOG_EXPIRED this morning around 6:30 ET. The node had been running for roughly 22 hours. I have not updated the node with last night's build yet. IP=52.206.43.80

Trying to run `venv/bin/python scripts/crontab.py` from the sentinel directory times out. It had timed out for at least 2 hours to become watchdog_expired.

This is happening repeatedly with no successful executions. I bounced the daemon but still no use. I am able to run local dash-cli commands. I am running the latest sentinel code from `https://github.com/nmarley/sentinel.git`

While writing this ticket, another this happened to a second node. All the same symptoms. IP=52.206.45.35.

<img width="933" alt="screen shot 2016-10-30 at 8 58 07 am" src="https://cloud.githubusercontent.com/assets/1169569/19836702/0480dd6a-9e7f-11e6-93e1-93ef3c2021aa.png">

**Comment by ryansd92** [comment]
Looking at the CPU usage, there were no spikes in the last 24 hours that would have prevented the execution of either sentinel or the daemon. Memory looks fine too.

**Comment by chaeplin** [comment]
caused by httplib. 
https://github.com/bitcoin/bitcoin/commit/ddf98d1d84343f2db0502a85628ef80f2ec57dbd

https://github.com/chaeplin/python-bitcoinrpc/commit/df4ba85b0db2f2fdeaf37b5f789bd18a05f08534

**Comment by ryansd92** [comment]
I'm happy to but I'm not sure I know how.

**Comment by ryansd92** [comment]
Ok it looks like this patch worked. Perfect.

From the sentinel directory, running the following command gets the latest code. Re-running sentinel then works without timing out.

`venv/bin/pip install --upgrade git+https://github.com/chaeplin/python-bitcoinrpc.git`

**Comment by ryansd92** [comment]
Re-opening this issue because the problem has persisted over the last day. I do not know if this important information but this instance of sentinel is not running on a masternode. It is running on a wallet. I have confirmed that the patch from yesterday is in place. I also removed and re-installed sentinel but the timeout continues.

Notice, that sentinel is able to connect to the daemon for at least `dash-cli mnsync status` but times out during it's normal interaction.

<img width="1124" alt="screen shot 2016-10-31 at 8 35 04 am" src="https://cloud.githubusercontent.com/assets/1169569/19854454/6950bd20-9f45-11e6-9cbd-f0363519e9e1.png">
